tools/profileBuilder/model/paths.go

```diff
@@ -0,0 +1,91 @@
+// +build go1.9
+
+// Copyright 2018 Microsoft Corporation and contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const pkgPathRegex = `services[/\\](?P<prv>[\w\-\._/\\]+)[/\\](?P<ver>v?\d{4}-\d{2}-\d{2}[\w\.\-]*|v?\d+[\.\w\-]*)[/\\](?P<grp>[/\\\w\-\._]+)`
+
+var (
+	validPackagePath   = regexp.MustCompile(pkgPathRegex)
+	validPkgPathModVer = regexp.MustCompile(pkgPathRegex + `[/\\](?P<mod>v\d+)[/\\]?(?P<api>\w+api)?`)
+	apiPkgRegex        = regexp.MustCompile(`[/\\]\w+api$`)
+)
+
+// PathInfo provides information about a package's path.
+type PathInfo struct {
+	IsArm    bool
+	Provider string
+	Version  string
+	Group    string
+	ModVer   string
+	APIPkg   string
+}
```

Comment:
sooo we are sure everything follows this convention ... do we check on the PRs to the repo ?

Reply:
Yes (pkgchk tool helps enforce this) however note that some items are not singular, here's an example.
`
D:\work\src\github.com\Azure\azure-sdk-for-go\services\datalake\analytics\mgmt\2016-11-01\account
`
This would result in the following struct values.
```go
PathInfo {
  IsArm: true,
  Provider: "datalake/analytics",
  Version: "2016-11-01",
  Group: "account",
}
```
There are similar cases where group is not singular (I believe cognitive services has several).